### PR TITLE
feat: stamp our version into release

### PR DESCRIPTION
**Pull request by alexeagle**
We can use this at runtime to support features like checking for updates.

### CODE REVIEW

File: .github/workflows/release_prep.sh [hunk 1]
```diff
@@ -9,6 +9,13 @@ TAG=${GITHUB_REF_NAME}
 PREFIX="rules_ts-${TAG:1}"
 ARCHIVE="rules_ts-$TAG.tar.gz"
 
+# Don't include examples in the distribution artifact, to reduce size
+# Note, we do need to include e2e/bzlmod since BCR runs our test based on distribution artifact.
+echo >>.git/info/attributes "examples export-ignore"
+
+# Stamp our release version into source files
+git grep -l '"0.0.0"' | xargs sed -i "s/0.0.0/${TAG:1}/"
```
Comment (gregmagolan) [inline .github/workflows/release_prep.sh:17]: hmmm. we may pickup `0.0.0` all over the place in lock files.. should we make it 0.0.0-PLACEHOLDER?

Comment (alexeagle) [reply .github/workflows/release_prep.sh:17]: in my testing, the double quotes prevented that and only matched on our files. But I'm investigating the `export-subst` feature of https://git-scm.com/docs/gitattributes#_creating_an_archive to make this more targeted

Comment (alexeagle) [reply .github/workflows/release_prep.sh:17]: The git version on my machine is old, so testing "live in prod" to be sure https://github.com/aspect-build/rules_esbuild/commit/a8c4379990372c48bc05aea608caf929e59b485d

Comment (alexeagle) [reply .github/workflows/release_prep.sh:17]: Current GHA runners have

```
Getting Git version info
  Working directory is '/home/runner/work/rules_esbuild/rules_esbuild'
  /usr/bin/git version
  git version 2.41.0
```

Comment (alexeagle) [reply .github/workflows/release_prep.sh]: Okay, verified this new approach:
https://github.com/aspect-build/rules_esbuild/releases/tag/v0.15.3

I had to upgrade `git` on my machine before I could make it work locally as well.
PTAL

**Review by kormide — APPROVED** [review]

**Comment by alexeagle** [comment]
Note: I'm not going to try to upstream this approach to the bazel-contrib/rules-template since it doesn't cover the root MODULE.bazel file and so there's still a patch required for the BCR, I think ppl will object and point out that the stamp mechanism should replace that file too.

**Pull request merged**